packages/storybook-vue/stories/2_for_developers/02.ScaleAndVue_en.md

```diff
@@ -1,59 +1,93 @@
 # Scale and Vue
 
-`Vue` works well with custom elements, but has some minor limitations.
+Vue has full support for Web Components.
 
-To work around this problems we provide custom element wrappers for better developer experience.
+You can find an example boilerplate app in the examples folder in the [GitHub repo](https://github.com/telekom/scale/tree/main/examples).
 
-Please note that you will need to install both, the `@telekom/scale-components` as well as the `@telekom/scale-components-vue` packages.
+## Setup
 
-Once you have installed both packages and defined the custom elements, `Scale` components behave mostly the way you would expect a `Vue` component to behave!
+First, install the package:
 
-## src/main.ts
+```bash
+npm install @telekom/scale-components
+```
+
+Load the component library and the CSS in `main.js`:
 
-```javascript
-import Vue from 'vue';
-import App from './App.vue';
-import {
-  applyPolyfills,
-  defineCustomElements,
-} from '@telekom/scale-components/loader';
+```js
+import { defineCustomElements } from '@telekom/scale-components/loader';
 import '@telekom/scale-components/dist/scale-components/scale-components.css';
 
-Vue.config.productionTip = false;
-Vue.config.ignoredElements = [/scale-\w*/];
+// ...
+
+defineCustomElements();
+```
+
+In order to instruct Vue to treat Scale components as custom elements, [a bit of configuration is needed](https://v3.vuejs.org/guide/migration/custom-elements-interop.html#autonomous-custom-elements).
 
-applyPolyfills().then(() => {
-  defineCustomElements(window);
-});
+For **Vue v2**:
 
-new Vue({
-  render: (h) => h(App),
-}).$mount('#app');
+```js
+// main.js
+Vue.config.ignoredElements = [/scale-\w*/];
+```
+
+For **Vue v3**, set `compilerOptions.isCustomElement`:
+
+```js
+// in webpack config
+rules: [
+  {
+    test: /\.vue$/,
+    use: 'vue-loader',
+    options: {
+      compilerOptions: {
+        isCustomElement: (tag) => tag.startsWith('scale-'),
+      },
+    },
+  },
+  // ...
+];
+
+// or in main.js (on-the-fly template compilation)
+const app = Vue.createApp({});
+app.config.compilerOptions.isCustomElement = (tag) => tag.startsWith('scale-');
 ```
```

Comment:
Hey,
It's a nice improvement for vue3.
But it only handles basic setups with Webpack and doesn't have a solution for vue-cli which uses vue.config.js files.
For our Project using single-spa-vue a vue.config.js file would be preferred since we would dislike using an additional webpack file.

Reply:
Hey, we can surely add the `vue.config.js` version of this, which is actually what we're using in the updated Vue3 example.

```js
// vue.config.js
  chainWebpack: config => {
    config.module
      .rule('vue')
      .use('vue-loader')
        .tap(options => {
          // https://v3.vuejs.org/guide/migration/custom-elements-interop.html#autonomous-custom-elements
          options.compilerOptions = {
            isCustomElement: (tag) => tag.startsWith('scale-')
          }
          return options
        })
  }
```

This is what you're referring to, right?

https://github.com/telekom/scale/blob/bd37956abfa678791073b1ebbf08937908e6502f/examples/vue3-boilerplate/vue.config.js

Reply:
Jup exactly 👍

Reply:
amazing, will update